src/app/api/main.py

```
@@ -116,12 +130,127 @@ async def lifespan(app: FastAPI):
+        workflows = app.state.workflows
+        await _execute_workflow(payload, workflows)
+        _set_status(job_id, JobStatus.SUCCESS)
+    except Exception as exc:
+        logger.error(f"Error while running job {job_id}: {exc}", exc_info=True)
+        _set_status(job_id, JobStatus.ERROR)
+        raise
+        
+        
+
+@app.post("/jobs", response_model=JobOut, status_code=201)
+async def create_or_update_job(job: JobIn,  session: Session = Depends(get_sqlmodel_session), _: SessionModel = Depends(verify_session_token)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)